File: TOMIMSV4/Entity/Mantenimientos/Acuerdos_Comerciales/Trans_AcuerdosComerciales_Det/clsBeTrans_acuerdoscomerciales_det.vb
Public Class clsBeTrans_acuerdoscomerciales_det
    Implements ICloneable

    Public Property IdAcuerdoDet() As Integer = 0
    Public Property IdAcuerdoEnc() As Integer = 0
    Public Property Codigo_producto() As String = ""
    Public Property Servicio() As String = ""
    Public Property Nemonico() As String = ""
    Public Property Codigo_acuerdo() As Integer = 0
    Public Property Correlativo_detalleacuerdo() As Integer = 0
    Public Property Descripcion() As String = ""
    Public Property Numero_unidades() As Double = 0.0
    Public Property Monto() As Double = 0.0
    Public Property Porcentaje() As Double = 0.0
    Public Property Dias_eventos() As Integer = 0
    Public Property Corre_cbcatalogoproductos() As Integer = 0
    Public Property Estado() As Boolean = False
    Public Property Prioridad() As Integer = 0
    Public Property User_agr As Integer = 0
    Public Property Fec_agr As DateTime = Now
    Public Property User_mod As Integer = 0
    Public Property Fec_mod As DateTime = Now
    Public Property IdBodega As Integer = 0
    Public Property IdTipoCobro As Integer = 0


    Sub New()
        '#EJC: Add your constructor here... 
    End Sub

    Public Function Clone() As Object Implements System.ICloneable.Clone
        Return MyBase.MemberwiseClone()
    End Function

End Class

File: WMS.EntityCore/Acuerdos/clsBeTrans_acuerdoscomerciales_det.cs
namespace WMS.EntityCore.AcuerdosComerciales
{
    public class clsBeTrans_acuerdoscomerciales_det : ICloneable
    {
        public int IdAcuerdoDet { get; set; } = 0;
        public int? IdAcuerdoEnc { get; set; } = null;
        public string Codigo_producto { get; set; } = "";
        public string Servicio { get; set; } = "";
        public string Nemonico { get; set; } = "";
        public int? Codigo_acuerdo { get; set; } = null;
        public int? Correlativo_detalleacuerdo { get; set; } = null;
        public string Descripcion { get; set; } = "";
        public decimal? Numero_unidades { get; set; } = 0;
        public decimal? Monto { get; set; } = 0;
        public decimal? Porcentaje { get; set; } = 0;
        public int? Dias_eventos { get; set; } = null;
        public int? Corre_cbcatalogoproductos { get; set; } = null;
        public bool? Estado { get; set; } = true;
        public byte? Prioridad { get; set; } = 0;
        public int? IdBodega { get; set; } = null;
        public int? IdTipoCobro { get; set; } = null;
        public int? User_agr { get; set; } = null;
        public DateTime? Fec_agr { get; set; } = DateTime.Now;
        public int? User_mod { get; set; } = null;
        public DateTime? Fec_mod { get; set; } = DateTime.Now;

        /// <summary>
        /// Campo adicional para control de envío a ERP
        /// </summary>
        public bool Enviado { get; set; } = false;

        public clsBeTrans_acuerdoscomerciales_det()
        {
        }

        public clsBeTrans_acuerdoscomerciales_det(
            int idAcuerdoDet,
            string codigo_producto,
            string descripcion,
            int? idAcuerdoEnc = null)
        {
            IdAcuerdoDet = idAcuerdoDet;
            Codigo_producto = codigo_producto ?? "";
            Descripcion = descripcion ?? "";
            IdAcuerdoEnc = idAcuerdoEnc;
        }

        public object Clone()
        {
            return this.MemberwiseClone();
        }
    }
}